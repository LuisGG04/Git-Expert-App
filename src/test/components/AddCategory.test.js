import React from "react"
import "@testing-library/jest-dom"
import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory"

describe('Test del componente AddCategory en sus funciones', () => {

    const setCategories = jest.fn()
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>)
    
    beforeEach(()=> {
        jest.clearAllMocks()
        wrapper = shallow(<AddCategory setCategories={setCategories}/>)
    })
    
    test('Debe de mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('Evaluación en la caja de texto', () => {
        const input = wrapper.find("input")
        const value = "Hola Mundo"
        input.simulate("change",{target:{value}})

        expect(wrapper.find("p").text().trim()).toBe(value)
    })

    test('Debe de mostrar el formulario sin postear la información', () => {
        wrapper.find("form").simulate("submit",{preventDefault(){}})
        expect(setCategories).not.toHaveBeenCalled()
    })
    
    test('Debe de llamar el setCategories y limpiar la caja de text', () => {
        const input = wrapper.find("input")
        const value = "Hola Mundo"
        input.simulate("change",{target:{value}})
        
        expect(wrapper.find("p").text().trim()).toBe(value)
        wrapper.find("form").simulate("submit",{preventDefault(){}})
        expect(setCategories).toHaveBeenCalled()

        expect(input.prop("value")).toBe("")
    })
    
})
