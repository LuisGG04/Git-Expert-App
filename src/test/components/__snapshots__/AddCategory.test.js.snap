// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test del componente AddCategory en sus funciones Debe de mostrar el componente correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
